lib/types.nix

```
@@ -625,6 +646,45 @@ rec {
     addCheck = elemType: check: elemType // { check = x: elemType.check x && check x; };
 
   };
+
+  /* Explicitly define a secret path in the world-readable Nix store, allowing
+     it to be used as a value for options of type secretPath.
+  */
+  secretInNixStore = path:
```

Some of us are conditioned to recognize that this is dangerous, but newcomers aren't.

```suggestion
  makeWorldReadable = path:
```

---

I originally didn't choose that because world-readable sounds like literally everybody can read it, but really it's just the people that have access to the machine.

Also, secrets don't have to be present on the evaluation host, in which case the secret can't be made world readable (though it already is on the remote host).

I think it's also important to clearly indicate what a function does, and this function does "Allow a secret path to be in the nix store, potentially importing it first". Though I like the idea of better indicating that it's it's not secure in general.

How about:
- `storeExposedSecret`
- `nixStoreExposedSecret`
- `unprotectedStoreSecret`
- `insecureStoreSecret`
- `secretInUnprotectedStore`
- `secretInUnprotectedNixStore`
- `exposedStoreSecret`

---

> in which case the secret can't be made world readable (though it already is on the remote host).

Judging from the examples in https://github.com/NixOS/nixpkgs/pull/78640#issuecomment-582063315, this seems to apply only to (iv). That one seems like a particularly inconvenient and insecure method.

A name like that seems ok. Which one is best may depend on whether or not the `{ _type = "secretPath"; }` idea is implemented.

---

I would like to see the word `insecure` here. `insecureStoreSecret` is good in my book